common/lib/container-definitions/src/loader.ts

```diff
@@ -93,6 +94,26 @@ export interface IContainerEvents extends IEvent {
     (event: "dirty" | "saved", listener: (dirty: boolean) => void);
 }
 
+/**
+ * Enum defining the different states of connectivity a container can be in
+ */
+ export enum ConnectionState {
```

Comment:
i'd prefer if we didn't add a new enum to the definition packages. I'd like to eventually get our defintion packages to only export types aka noting the exports javascript code. instead of an enum, i would prefer this:

```typescript
export namespace ConnectionState {
    /**
     * The document is no longer connected to the delta server
     */
    export type Disconnected = 0;

    /**
     * The document has an inbound connection but is still pending for outbound deltas
     */
    export type Connecting =1;

    /**
     * The document is fully connected
     */
     export type Connected =2;
}
export type ConnectionState = ConnectionState.Disconnected | ConnectionState.Connecting | ConnectionState.Connected;
```

Doing this will still require the implementor of anything using ConnectionState to provide the value, as this definitions will only specify the types (shape)

Reply:
i also pefer this, as typescript doesn't actually recommend using enums. they are one of the few features that doesn't natively exist in javascript.

I use a similar pattern for [SummaryType](https://github.com/microsoft/FluidFramework/blob/f2f0d6147b48bf99731e7425a97db6c9681b1f19/common/lib/protocol-definitions/src/summary.ts#L24). This case also provides the value, but that was because this was an enum conversation, not an addition, eventually i'd like to remove the values

Reply:
Updated the enum to now be a type as recommended. Thanks!